jwtauthccl: TestSubjectReservedUser is flaky

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/9370305?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
panic: test timed out after 55s
```

goroutine dump at: https://gist.github.com/rafiss/0614a4a354a95183dc3fdbbb7438c49a

cc @kpatron-cockroachlabs for awareness, though probably another team is going to have to fix this, since it looks like some internal queries were hanging.

--- a/pkg/upgrade/upgrades/system_job_info.go
+++ b/pkg/upgrade/upgrades/system_job_info.go
@@ -14,18 +14,40 @@ import (
 	"context"
 
 	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/systemschema"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
 	"github.com/cockroachdb/cockroach/pkg/upgrade"
 )
 
-// systemPrivilegesTableMigration creates the system.privileges table.
+// systemPrivilegesTableMigration creates the system.job_info table.
 func systemJobInfoTableMigration(
 	ctx context.Context, _ clusterversion.ClusterVersion, d upgrade.TenantDeps,
 ) error {
-	return createSystemTable(
+	// Create the job_info table proper.
+	if err := createSystemTable(
 		ctx, d.DB.KV(), d.Settings, d.Codec, systemschema.SystemJobInfoTable,
-	)
+	); err != nil {
+		return err
+	}
+
+	// Also bump the version of the descriptor for system.jobs. This
+	// ensures that this migration synchronizes with all other
+	// migrations that touch jobs, and concurrent transactions that
+	// perform DDL operations. We need this because just creating a new
+	// table above does not synchronize with other accesses to
+	// system.jobs on its own.
+	return d.DB.DescsTxn(ctx, func(
+		ctx context.Context, txn descs.Txn,
+	) (err error) {
+		collection := txn.Descriptors()
+		t, err := collection.MutableByID(txn.KV()).Desc(ctx, keys.JobsTableID)
+		if err != nil {
+			return err
+		}
+		return collection.WriteDesc(ctx, true /* kvTracing */, t, txn.KV())
+	})
 }
 
 const alterPayloadToNullableQuery = `

--- a/pkg/upgrade/upgrades/system_job_info_test.go
+++ b/pkg/upgrade/upgrades/system_job_info_test.go
@@ -16,12 +16,15 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/server"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/desctestutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/upgrade/upgrades"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/stretchr/testify/require"
 )
 
 func TestSystemJobInfoMigration(t *testing.T) {
@@ -51,9 +54,22 @@ func TestSystemJobInfoMigration(t *testing.T) {
 	db := tc.ServerConn(0)
 	defer db.Close()
 
-	// NB: this isn't actually doing anything, since the table is baked into the
-	// bootstrap schema, so this is really just showing the upgrade is idempotent,
-	// but this is in line with the other tests of createSystemTable upgrades.
+	// Advance to the version immediately before the one that
+	// interests us.
+	upgrades.Upgrade(
+		t,
+		db,
+		clusterversion.V23_1CreateSystemJobInfoTable-1,
+		nil,
+		false,
+	)
+
+	// We verify that the jobs table gets its version upgraded through
+	// the upgrade, to ensure the creation of job_info synchronizes with
+	// concurrent accesses to the jobs table.
+	kvDB := tc.Server(0).(*server.TestServer).DB()
+	tblBefore := desctestutils.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "system", "public", "jobs")
+
 	upgrades.Upgrade(
 		t,
 		db,
@@ -61,4 +77,7 @@ func TestSystemJobInfoMigration(t *testing.T) {
 		nil,
 		false,
 	)
+
+	tblAfter := desctestutils.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "system", "public", "jobs")
+	require.Greater(t, tblAfter.GetVersion(), tblBefore.GetVersion())
 }

